client/comms/wsconn.go

```diff
@@ -101,6 +107,7 @@ func NewWsConn(cfg *WsCfg) (WsConn, error) {
 		tlsConfig = &tls.Config{
 			RootCAs:    rootCAs,
 			MinVersion: tls.VersionTLS12,
+			ServerName: uri.Hostname(),
```

Comment:
I'm surprised this is needed for websocket clients because gorilla's dialer sets the ServerName from the hostname.  @vctt94 Any chance you can test this PR since connecting to dex-test.ssgen.io only worked via IP not hostname?

Reply:
Since this solved your issue @vctt94 that suggests a gorilla bug that was improperly setting ServerName in the Dialer.  I suspect your DNS is resolving to ipv6 and that's messing up gorilla's ServerName setting from the internal hostNoPort variable.  I'd be willing to bet there's a bug in there.

Reply:
Do you know of any reason that we **should not** set the `ServerName` manually?

Reply:
No, it's just curious that we have to since gorilla should do it.  Or rather there's some edge case with @vctt94 that makes us have to.